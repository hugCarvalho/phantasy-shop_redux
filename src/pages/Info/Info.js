import React from "react";
import { listOfTasks } from "./listOfTasks";
import "./Info.scss";

//TODO: move fns to separate file
//FNS
const taskStatus = (status) => {
  if (status === true) return { color: "green" };
  else if (!status) return { color: "red" };
  else return { color: "orange" };
};

//Computation for report
const calcCategories = (list) => {
  const firstWordOfEachTitle = list.map((item) => item.title.split(" ")[0]);
  const categories = {};

  firstWordOfEachTitle.forEach((item) => {
    item = item.slice(0, item.length - 1); //removes colon
    if (!categories[item]) return (categories[item] = 1);
    else return (categories[item] += 1);
  });
  return Object.entries(categories);
};
const calcCompleted = (list) => list.filter((item) => item.completed).length;
const calcCompletedPercent = (list) =>
  Math.round((calcCompleted(list) * 100) / list.length);
//-FNS END-//

const ongoingText = `ONGOING project, NOT fully optimized.`;
//Some features are still being added

function Home() {
  const [{ content, cursor: index }, setContent] = React.useState({
    content: "",
    cursor: 0,
  });

  React.useEffect(() => {
    if (index === ongoingText.length) {
      return;
    }

    const delay = setTimeout(() => {
      setContent({ content: content + ongoingText[index], cursor: index + 1 });
    }, 0 || 140);

    return () => {
      clearTimeout(delay);
    };
  }, [content, index]);

  return (
    <div className="Home">
      <article>
        <h1>Important!</h1>
        <h3 className="typing">
          {" "}
          <span>{content}</span>
        </h3>
        <hr></hr>

        <article className="built-with">
          <strong>
            Built with{" "}
            <span className="bb-style">
              REACT <sup>hooks</sup>
            </span>{" "}
            + <span className="bb-style">REDUX</span> +{" "}
            <span className="bb-style">SASS</span>
          </strong>
        </article>

        <article>
          <p>
            This project simulates a <b>phantasy online shop</b> where you can book days
            with actors of the Breaking Bad series.
          </p>

          <p>
            The data consists of a mixture of the{" "}
            <span className="bb-style">
              Breaking Bad <b>API</b>
            </span>{" "}
            data and own object manipulation in order to add the necessary entries for a
            shop behaviour.
          </p>
        </article>
      </article>
      <hr></hr>

      <div className="legend">
        <h5>Tasks</h5>
        <div className="wrapper">
          <span style={{ color: "green" }}>green: completed</span>
          <span style={{ color: "orange" }}>orange: partially completed</span>
          <span style={{ color: "red" }}>red: not began</span>
        </div>
      </div>
      <hr></hr>

      <section className="list">
        <h5>List</h5>
        <ul>
          {listOfTasks.map((item, i) => (
            <li key={i} style={taskStatus(item.completed)}>
              {item.title}
            </li>
          ))}
        </ul>
      </section>

      <section className="statistics">
        <h5>Report</h5>
        <p>{`There are a total of ${listOfTasks.length} tasks, from which ${calcCompleted(
          listOfTasks
        )} are completed (${calcCompletedPercent(listOfTasks)}%)  `}</p>
        <p>{`There are ${calcCategories(listOfTasks).length} categories:`}</p>
        {calcCategories(listOfTasks).map((item, i) => (
          <p key={i}>
            {item[0]} : {item[1]} {item[1] > 1 ? "tasks" : "task"}
          </p>
        ))}
      </section>
    </div>
  );
}

export default Home;
