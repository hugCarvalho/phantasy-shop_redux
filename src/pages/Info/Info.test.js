import React from "react";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import Info from "./Info";

Enzyme.configure({ adapter: new Adapter() });

test("make snapshot component", () => {
  expect(shallow(<Info />)).toMatchSnapshot();
});

test("snapshot", () => {
  const tree = renderer.create(<Info />).toJSON();
  expect(tree).toMatchSnapshot();
});
