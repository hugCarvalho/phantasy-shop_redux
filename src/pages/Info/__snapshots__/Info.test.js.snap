// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`make snapshot component 1`] = `ShallowWrapper {}`;

exports[`snapshot 1`] = `
<div
  className="Home"
>
  <article>
    <h1>
      Important!
    </h1>
    <h3
      className="typing"
    >
       
      <span>
        
      </span>
    </h3>
    <hr />
    <article
      className="built-with"
    >
      <strong>
        Built with
         
        <span
          className="bb-style"
        >
          REACT 
          <sup>
            hooks
          </sup>
        </span>
         
        + 
        <span
          className="bb-style"
        >
          REDUX
        </span>
         +
         
        <span
          className="bb-style"
        >
          SASS
        </span>
      </strong>
    </article>
    <article>
      <p>
        This project simulates a 
        <b>
          phantasy online shop
        </b>
         where you can book days with actors of the Breaking Bad series.
      </p>
      <p>
        The data consists of a mixture of the
         
        <span
          className="bb-style"
        >
          Breaking Bad 
          <b>
            API
          </b>
        </span>
         
        data and own object manipulation in order to add the necessary entries for a shop behaviour.
      </p>
    </article>
  </article>
  <hr />
  <div
    className="legend"
  >
    <h5>
      Tasks
    </h5>
    <div
      className="wrapper"
    >
      <span
        style={
          Object {
            "color": "green",
          }
        }
      >
        green: completed
      </span>
      <span
        style={
          Object {
            "color": "orange",
          }
        }
      >
        orange: partially completed
      </span>
      <span
        style={
          Object {
            "color": "red",
          }
        }
      >
        red: not began
      </span>
    </div>
  </div>
  <hr />
  <section
    className="list"
  >
    <h5>
      List
    </h5>
    <ul>
      <li
        style={
          Object {
            "color": "orange",
          }
        }
      >
        Router: Setting React-router logic for page navigation
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Async: fetching logic for fetching data from Breaking Bad API 
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Async: Adapting received data to online shop concept by adding entries to it, amount, price, etc
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Itemlist: Displaying data in item overview
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Itemlist: Add \`add to cart / remove from cart logic\`
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: rate actor. Only if logged in
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: add to favorites. Must be logged in
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: pagination
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: filters
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Itemlist: item discount - 2 prices: old & new
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Itemlist: format currency
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: displaying number of products in cart on the fly 
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: add cart icon
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: display items
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: increase / decrease quantities 
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: calc item subtotal + calc cart total
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Cart: add inventory logic
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Cart: add tax + item discount + cart discount + enter voucher
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Cart: add buy button and mockup behaviour(is login success else ask for login)
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Register: do register form - html + css
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Register: add logic - only frontend sim
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Login: do login form - html + css 
      </li>
      <li
        style={
          Object {
            "color": "green",
          }
        }
      >
        Login: add logic to form - only frontend sim
      </li>
      <li
        style={
          Object {
            "color": "orange",
          }
        }
      >
        Alerts: to be defined
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        Responsive: add media queries
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Item List
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Cart
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Register
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Login
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Nav Bar
      </li>
      <li
        style={
          Object {
            "color": "red",
          }
        }
      >
        CSS: make things prettier - Home
      </li>
    </ul>
  </section>
  <section
    className="statistics"
  >
    <h5>
      Report
    </h5>
    <p>
      There are a total of 31 tasks, from which 17 are completed (55%)  
    </p>
    <p>
      There are 9 categories:
    </p>
    <p>
      Router
       : 
      1
       
      task
    </p>
    <p>
      Async
       : 
      2
       
      tasks
    </p>
    <p>
      Itemlist
       : 
      8
       
      tasks
    </p>
    <p>
      Cart
       : 
      8
       
      tasks
    </p>
    <p>
      Register
       : 
      2
       
      tasks
    </p>
    <p>
      Login
       : 
      2
       
      tasks
    </p>
    <p>
      Alerts
       : 
      1
       
      task
    </p>
    <p>
      Responsive
       : 
      1
       
      task
    </p>
    <p>
      CSS
       : 
      6
       
      tasks
    </p>
  </section>
</div>
`;
